AK/IntegralMath.h

```diff
@@ -54,4 +54,18 @@ constexpr I pow(I base, I exponent)
     return res;
 }
 
+template<auto base, Unsigned U = decltype(base)>
+constexpr bool is_power_of(U x)
+{
+    if constexpr (base == 2)
+        return is_power_of_two(x);
+
+    // FIXME: I am naive! A log2-based approach (pow<U>(base, (log2(x) / log2(base))) == x) does not work due to rounding errors.
```

Comment:
This should definitely have a test exposing the issue you have solved by implementing it naively, to prevent someone else from breaking it in the future.